### ISSUE TITLE
OpenCL file support

### ISSUE THREAD

**Issue by MichaelHancock** [Glavin001/atom-beautify#870]
### Description

Beautification does not work at all on opencl files. Even though said files are c based 

Here is an example of some of the code i would like to beautify:

```
__kernel void reduce_max(__global const int* A, __global int* B, __local int* scratch) {
    int id = get_global_id(0);
    int lid = get_local_id(0);
    int N = get_local_size(0);

    scratch[lid] = A[id];

    barrier(CLK_LOCAL_MEM_FENCE);

    for (int i = 1; i < N; i *= 2) {
        if (!(lid % (i * 2)) && ((lid + i) < N))
        {
            if (scratch[lid] < scratch[lid + i])
                scratch[lid] = scratch[lid+i];
        }
        barrier(CLK_LOCAL_MEM_FENCE);
    }

    //copy the cache to output array
    if (!lid) {
        atomic_max(&B[0],scratch[lid]);
    }

}
```

Since this code is c based, you could have beautify identify .cl  as it does for C files.
### Steps to Reproduce

1. Add code shown above to Atom editor
2. Run command `Atom Beautify: Beautify Editor`
3. Notice that it wont work!

**Comment by prettydiff** [comment]
Would you mind posting the debug.md file to a gist?  You can generate this file by running Packages -> Atom Beautify -> Debug

**Comment by MichaelHancock** [comment]
Hi,  i would be happy to do that! But when i follow the steps provided i get the follwoing error...

```
Uncaught TypeError: Cannot read property 'name' of null
/C:/Users/Michael%20Hancock/.atom/packages/atom-beautify/src/beautify.coffee:308

TypeError: Cannot read property 'name' of null
    at atom-workspace.debug (file:///C:/Users/Michael Hancock/.atom/packages/atom-beautify/src/beautify.coffee:346:51)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (C:\Users\Michael Hancock\AppData\Local\atom\app-1.6.0\resources\app.asar\src\command-registry.js:260:29)
    at CommandRegistry.handleCommandEvent (C:\Users\Michael Hancock\AppData\Local\atom\app-1.6.0\resources\app.asar\src\command-registry.js:3:61)
    at CommandRegistry.module.exports.CommandRegistry.dispatch (C:\Users\Michael Hancock\AppData\Local\atom\app-1.6.0\resources\app.asar\src\command-registry.js:160:19)
    at AtomEnvironment.module.exports.AtomEnvironment.dispatchApplicationMenuCommand (C:\Users\Michael Hancock\AppData\Local\atom\app-1.6.0\resources\app.asar\src\atom-environment.js:1013:28)
    at emitOne (events.js:77:13)
    at EventEmitter.emit (events.js:169:7)
```

**Comment by Glavin001** [comment]
Bummer. The error is likely caused by line https://github.com/Glavin001/atom-beautify/blob/master/src/beautify.coffee#L346 

```coffeescript
  beautifiers = beautifier.getBeautifiers(language.name)
```

Specifically `language` being null and calling `language.name`.

---

I think the culprit would then be that the OpenCL file is not being detected as C by Atom (of course), so the language is null and Atom Beautify does not know what to do with it since the extension is not explicitly supported.

@MichaelHancock could you confirm the extension is `.cl` (instead of `.c`) and also is there a grammar being detected?
A simple Pull Request changing https://github.com/Glavin001/atom-beautify/blob/master/src/languages/c.coffee#L9-L18 will easily add OpenCL support :+1:.

- [ ] Add `.cl` extension for OpenCL
- [ ] Add grammar (is there an existing language grammar Atom detects?)

**Comment by MichaelHancock** [comment]
I will make that pull request now. I am not 100% on the grammar though. To make Atom detect opencl at all: https://github.com/podgib/atom-opencl . However i can confirm, for all opencl files, to be recognised at all as parallel kernel's, they must be saved as .cl.

**Comment by Glavin001** [comment]
Looking at atom-opencl I think the grammar could be `opencl`. You should be able to tell by looking at the bottom of your Atom Editor.

**Comment by Glavin001** [comment]
Closed by #872.

**Comment by Glavin001** [comment]
Published to v0.28.28
